aten/src/ATen/native/cuda/AdaptiveAveragePooling.cu

```diff
@@ -757,6 +757,7 @@ namespace {
     const Tensor& gradOutput,
     const Tensor& input)
   {
+    globalContext().alertNotDeterministic("adaptive_avg_pool2d_backward_out_cuda");
```

Comment:
Looking at these alerts, it would probably useful if we had "receipts" for why the are not deterministic (some sort of comment saying something like `// Non-deterministic because of use of atomicAdd below`). If these ever go out of date, the receipt would help a future developer understand when the warning could be removed.

Reply:
Good idea, I'll add those.

Just a thought--would it be useful to print out this information as part of the alert? I could potentially add a `reason` argument to `alertNotDeterministic`. Perhaps this information wouldn't be useful for the user though.